src/overlays/actors/ovl_En_Stream/z_en_stream.c

```diff
@@ -29,19 +31,117 @@ const ActorInit En_Stream_InitVars = {
+#endif
+
+void EnStream_SuckPlayer(EnStream* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad48;
+    Vec3f posRotDifference;
+    f32 xzDist;
+    f32 yDistWithOffset;
+    s32 pad30;
+    s32 pad2C;
+
+    if (func_80B0B81C(&this->actor.posRot, &player->actor.posRot, &posRotDifference, this->actor.scale.y) != 0) {
+        xzDist = sqrtf(SQ(posRotDifference.x) + SQ(posRotDifference.z));
+        yDistWithOffset = player->actor.posRot.pos.y - (this->actor.posRot.pos.y - 90.0f);
+        player->windDirection = Math_atan2f(-posRotDifference.x, -posRotDifference.z) * 10430.378f;
+        if (xzDist > 3.0f) {
+            Math_SmoothScaleMaxMinF(&player->windSpeed, 3.0f, 0.5f, xzDist, 0.0f);
+        } else {
+            player->windSpeed = 0.0f;
+            Math_SmoothScaleMaxMinF(&player->actor.posRot.pos.x, this->actor.posRot.pos.x, 0.5f, 3.0f, 0.0f);
+            Math_SmoothScaleMaxMinF(&player->actor.posRot.pos.z, this->actor.posRot.pos.z, 0.5f, 3.0f, 0.0f);
+        }
+        if (yDistWithOffset > 0.0f) {
+            Math_SmoothScaleMaxMinF(&player->actor.velocity.y, -3.0f, 0.7f, yDistWithOffset, 0.0f);
+            if (posRotDifference.y < -70.0f) {
+                player->stateFlags2 |= 0x80000000;
+            }
+        }
+    } else {
+        EnStream_SetupAction(this, &EnStream_WaitForPlayer);
+    }
+}
+
+void EnStream_WaitForPlayer(EnStream* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 pad;
+    Vec3f temp;
+
+    if (func_80B0B81C(&this->actor.posRot, &player->actor.posRot, &temp, this->actor.scale.y) != 0) {
+        EnStream_SetupAction(this, &EnStream_SuckPlayer);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/func_80B0B934.s")
+void EnStream_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnStream* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/func_80B0BAC8.s")
+    this->actionFunc(this, globalCtx);
+    func_8002F948(thisx, NA_SE_EV_WHIRLPOOL - SFX_FLAG);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/EnStream_Update.s")
+void EnStream_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    u32 multipliedFrames;
+    u32 frames = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Stream/EnStream_Draw.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_stream.c", 295);
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_stream.c", 299),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    multipliedFrames = frames * 20;
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, frames * 30, -multipliedFrames, 0x40, 0x40, 1,
+                                multipliedFrames, -multipliedFrames, 0x40, 0x40));
+    gSPDisplayList(POLY_XLU_DISP++, &D_06000950);
```

Comment:
After treating D_06000950 as an array (see the previous comment), the `&` should be removed